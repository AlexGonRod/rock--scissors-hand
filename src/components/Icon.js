import React from 'react'

const Icon = ({ onClick, dataId, className }) => {
	return (
		<div
			data-id={dataId}
			onClick={onClick}
			className={className}
		></div>
	)
}

export default Icon
