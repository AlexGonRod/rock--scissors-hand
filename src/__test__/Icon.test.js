import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'
import Icon from '../components/Icon';

describe('<Icon />', () => {
    let wrapper = shallow(<Icon />);
    beforeEach(() => {
        wrapper = shallow(<Icon />);
    });
    test('<Icon /> renders', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('<Icon /> className prop', () => {
        const className='test-class';
        
        const wrapper = shallow(<Icon className={className}/>)
        expect(wrapper.find('.test-class').length).toEqual(1);
    })
    test('<Icon /> onClick prop', () => {
        let state= false;
        const className='test-class';

        const wrapper = shallow(<Icon className={className} onClick={() => {state = true} }/>)
        const button = wrapper.find('.test-class').simulate('click');
        expect(state).toEqual(true);
    })
});