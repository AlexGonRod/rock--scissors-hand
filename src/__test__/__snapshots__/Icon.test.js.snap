// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> <Icon /> renders 1`] = `<div />`;
